### DISCUSSION TITLE
Future of `shouldUnregister`

### DISCUSSION THREAD

**Discussion by VictorGosse** [react-hook-form/react-hook-form#11884] [Q&A]
Hello,

First of all, thanks for this library that we are using a lot in our project and that we love.

**My question**: will `shouldUnregister` disappear in the future ?

A bit of context:
- Before the last major update we were relying a lot on the fact that `true` was the default value of `shouldUnregister`. With v7, the default value has changed to `false` which introduced a lot of issues in our codebase so we forced it to `true` everywhere to ease the migration.
- As v7 is now mainly based on the fact that the default value is `false`, there are some features that we can't use properly as we have `shouldUnregister: true` in our codebase.
- I read here and there (eg: https://github.com/react-hook-form/react-hook-form/issues/10390#issuecomment-1547019679) that this option may become deprecated.

If it becomes deprecated, we'll have to update a big part of our codebase and we would like to do it as early as possible to ease the migration.

**Comment by zyhou** [comment]
@bluebill1049 any comment?
